pkg/agent/plugin/workloadattestor/docker/docker_posix.go

```diff
@@ -19,36 +23,67 @@ type OSConfig struct {
 	// ContainerIDCGroupMatchers is a list of patterns used to discover container IDs from cgroup entries.
 	// See the documentation for cgroup.NewContainerIDFinder in the cgroup subpackage for more information. (Unix)
 	ContainerIDCGroupMatchers []string `hcl:"container_id_cgroup_matchers" json:"container_id_cgroup_matchers"`
+
+	// UseNewContainerLocator, if true, uses the new container locator
+	// mechanism instead of cgroup matchers. Currently defaults to false if
+	// unset. This will default to true in a future release. (Unix)
+	UseNewContainerLocator *bool `hcl:"use_new_container_locator"`
+
+	// Used by tests to use a fake /proc directory instead of the real one
+	rootDir string
 }
 
-func createHelper(c *dockerPluginConfig) (*containerHelper, error) {
-	var containerIDFinder cgroup.ContainerIDFinder = &defaultContainerIDFinder{}
-	var err error
-	if len(c.ContainerIDCGroupMatchers) > 0 {
+func createHelper(c *dockerPluginConfig, log hclog.Logger) (*containerHelper, error) {
+	var containerIDFinder cgroup.ContainerIDFinder
+
+	switch {
+	case c.UseNewContainerLocator != nil && *c.UseNewContainerLocator:
```

Comment:
I think that allowing using the same variable can be confusing, if we want to allow users to add new matchers for this new locator, we may do in a separated variable,
so they know that old matchers will not work with new locator... 

but we really need to add matchers for this?

Reply:
My thought is that:
1. You won't set use_new_container_locator to true while still using the old matcheres
2. When we change the default of use_new_container_locator to true, we'll still fall back to the matchers, unless use_new_container_locator has been explicitly set to true.
3. The hope is that group matchers are no longer needed with the new container locator logic. If we've missed a use case, it would be good to know sooner than later.

After typing this out, I realize that it might be better to fail the configure step if you've explicitly set use_new_container_locator to true but also defined custom matchers. They should be mutually exclusive.

Reply:
I like the mutually exclusive, and agree for now we can avoid custom matchers, and add if users report it is required